### PR TITLE
Use nvim_echo() instead of calling echohl

### PR DESCRIPTION
**Pull request by delphinus**
Now Neovim head has `nvim_echo()` to show command messages with highlighting.

NOTE: It cannot deal with the `teasing` option from `refine_hints()`.

### CODE REVIEW

File: lua/hop/init.lua [hunk 1]
```diff
@@ -19,9 +19,10 @@ local function get_command_opts(local_opts)
 end
 
 -- Display error messages.
-local function eprintln(opts, msg)
-  if opts.teasing then
-    vim.cmd(string.format('echohl Error|echo "%s"', msg))
+local function eprintln(msg, opts)
+  local local_opts = get_command_opts(opts)
```
Comment (phaazon) [inline lua/hop/init.lua:23]: What do you need to do that? `get_command_opts` is only used at the top-level of the call-stack (i.e. in public functions). The options are then forwarded down to everything. If it’s due to `refine_hints`, we can simply modify `refine_hints` to get the options forwarded.

**Review by phaazon — COMMENTED** [review]
Thanks! Just as mall nits and that should be mergeable.

File: lua/hop/init.lua (hunk 1 above)
Comment (delphinus) [reply lua/hop/init.lua:23]: > `get_command_opts` is only used at the top-level of the call-stack

It is the point I also was worried about. `refine_hints` will be called from mappings, so cannot accept `opts` as it is.

https://github.com/phaazon/hop.nvim/blob/604e2e8138b36e4bf7e4a5a7562b483d3a617b51/lua/hop/keymap.lua#L21

> we can simply modify refine_hints to get the options forwarded.

Then can I change arguments for funcs as below?

```lua
-- signatures
local function eprintln(msg, teasing)

function M.refine_hints(buf_handle, key, teasing)

-- keymap definition
local teasing_str = opts.teasing and 'true' or 'false'
vim.api.nvim_buf_set_keymap(
  buf_handle,
  '',
  key,
  "<cmd>lua require'hop'.refine_hints(0, '" .. key .. "', " .. teasing_str .. ")<cr>",
  { nowait = true }
)
```

Comment (phaazon) [reply lua/hop/init.lua:23]: I think this concern is going to be addressed as soon as we merge #41 – i.e. it’s going to simplify your PR :laughing:

**Review by phaazon — CHANGES_REQUESTED** [review]
Can you please not merge `master` into your branch, [as explained here](https://github.com/phaazon/hop.nvim/blob/master/CONTRIBUTING.md#hygiene).

**Comment by delphinus** [comment]
Rebased. Check it :) @phaazon

**Pull request merged**